src/Wallabag/CoreBundle/Controller/SiteCredentialController.php

```diff
@@ -0,0 +1,156 @@
+     */
+    public function editAction(Request $request, SiteCredential $siteCredential)
+    {
+        $this->checkUserAction($siteCredential);
+
+        $deleteForm = $this->createDeleteForm($siteCredential);
+        $editForm = $this->createForm('Wallabag\CoreBundle\Form\Type\SiteCredentialType', $siteCredential);
+        $editForm->handleRequest($request);
+
+        if ($editForm->isSubmitted() && $editForm->isValid()) {
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($siteCredential);
+            $em->flush();
+
+            $this->get('session')->getFlashBag()->add(
+                'notice',
+                $this->get('translator')->trans('flashes.site_credential.notice.updated', ['%host%' => $siteCredential->getHost()])
+            );
+
+            return $this->redirectToRoute('site_credentials_edit', array('id' => $siteCredential->getId()));
+        }
+
+        return $this->render('WallabagCoreBundle:SiteCredential:edit.html.twig', array(
+            'credential' => $siteCredential,
+            'edit_form' => $editForm->createView(),
+            'delete_form' => $deleteForm->createView(),
+        ));
+    }
+
+    /**
+     * Deletes a site credential entity.
+     *
+     * @Route("/{id}", name="site_credentials_delete")
+     * @Method("DELETE")
+     */
+    public function deleteAction(Request $request, SiteCredential $siteCredential)
+    {
+        $this->checkUserAction($siteCredential);
+
+        $form = $this->createDeleteForm($siteCredential);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get('session')->getFlashBag()->add(
+                'notice',
+                $this->get('translator')->trans('flashes.site_credential.notice.deleted', ['%host%' => $siteCredential->getHost()])
+            );
+
+            $em = $this->getDoctrine()->getManager();
+            $em->remove($siteCredential);
+            $em->flush();
+        }
+
+        return $this->redirectToRoute('site_credentials_index');
+    }
+
+    /**
+     * Creates a form to delete a site credential entity.
+     *
+     * @param SiteCredential $siteCredential The site credential entity
+     *
+     * @return \Symfony\Component\Form\Form The form
+     */
+    private function createDeleteForm(SiteCredential $siteCredential)
+    {
+        return $this->createFormBuilder()
+            ->setAction($this->generateUrl('site_credentials_delete', array('id' => $siteCredential->getId())))
+            ->setMethod('DELETE')
+            ->getForm()
+        ;
+    }
+
+    /**
+     * Check if the logged user can manage the given site credential.
+     *
+     * @param SiteCredential $siteCredential The site credential entity
+     */
+    private function checkUserAction(SiteCredential $siteCredential)
+    {
+        if (null === $this->getUser() || $this->getUser()->getId() != $siteCredential->getUser()->getId()) {
```

Comment:
The item isn't visible in the menu is the paywall feature is disabled.
Maybe, instead of adding a hard check, I can just display a banner telling the feature is disabled and performing change will do nothing